games/static/games/proto.js

```diff
@@ -67,31 +107,52 @@ class Board{
         this.width = width
         this.aliens = []
         this.squares = []
-        const alienInvaders = [
-        0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
-        15, 16, 17, 18, 19, 20, 21, 22, 23, 24,
-        30, 31, 32, 33, 34, 35, 36, 37, 38, 39 
-        ]    
+        
+    }
+    createChildSquareAddToSquares(index) {
+        let div = document.createElement('div') 
+        this.parent.appendChild(div)
+        this.squares.push(new Square(index, div, this.width)) 
+        // return ?
+    }
+    draw(setup) {
+        let alienLocations = setup.getAlienLocations()
         for(let i = 0; i < width*width; i++) {
-            let div = document.createElement('div') 
-            this.parent.appendChild(div)
-            this.squares.push(new Square(i, div)) 
-
-            if (alienInvaders.includes(i)) {
-                let invader = createInvader(alienInvaders[i], width)
-                invader.draw(square)
-                this.aliens.push(invader)
-            } else if (i > width*width-width) {
-                square.classList.add('ground')
-            }
+            this.createChildSquareAddToSquares(i)
+        }
+        if(alienLocations.includes(i)) {
+            let invader = createInvader(alienLocations[i], width)
+            this.squares[i].setCharacter(invader)
+        } else if (i > width*width-width) {
```

Comment:
2 things:
1) level of abstraction - the draw function talks in high level, suddenly seeing `i > width*width-width` looks weird.
take it out to a function and give it a proper name.
we dont really care to have one line function as long as the name gives me enough information about the meaning of `i > width*width-width` (what does it symbolizes?)

also see this: `this.squares[i].div.classList.add('ground')` - we have divs, css classes, lists... should be abstracted away. try to think whos this responsibilty belongs to? i think just my reading that line again you can figure it out.

2) now, it feels that this `if` belongs somewhere else: when you create a child square you already have that knowledge. try to think about it